File: core/src/main/java/ir/logicfan/core/util/FileUtils.java
package ir.logicfan.core.util;

import android.content.Context;

import javax.inject.Singleton;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

@Singleton
public class FileUtils {

    public static String readTextFileFromAssets(Context context, String path) throws IOException {
        InputStream inputStream = context.getResources().getAssets().open(path);
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
        StringBuilder jsonString = new StringBuilder();
        String line;
        while ((line = bufferedReader.readLine()) != null) {
            jsonString.append(line);
        }
        inputStream.close();
        bufferedReader.close();
        return jsonString.toString();
    }
}

File: core/src/main/java/ir/logicbase/core/data/util/FileUtils.kt
package ir.logicbase.core.data.util

import android.content.Context
import java.io.IOException

object FileUtils {

    @JvmStatic
    @Throws(IOException::class)
    fun readTextFileFromAssets(context: Context, path: String): String =
        context.resources.assets.open(path).bufferedReader().use {
            it.readText()
        }
}
